source/fuzz/fuzzer_pass_inline_functions.cpp

```diff
@@ -0,0 +1,109 @@
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, transformation_context, fuzzer_context,
+                 transformations) {}
+
+FuzzerPassInlineFunctions::~FuzzerPassInlineFunctions() = default;
+
+void FuzzerPassInlineFunctions::Apply() {
+  std::vector<opt::Instruction*> function_call_instructions;
+
+  for (auto& function : *GetIRContext()->module()) {
+    for (auto& block : function) {
+      for (auto& instruction : block) {
+        if (!GetFuzzerContext()->ChoosePercentage(
+                GetFuzzerContext()->GetChanceOfInliningFunction())) {
+          continue;
+        }
+
+        // |instruction| must be OpFunctionCall to consider applying the
+        // transformation.
+        if (instruction.opcode() != SpvOpFunctionCall) {
+          continue;
+        }
+
+        // |called_function| must not have an early return.
+        auto called_function = fuzzerutil::FindFunction(
+            GetIRContext(), instruction.GetSingleWordInOperand(0));
+        if (called_function->HasEarlyReturn()) {
+          continue;
+        }
+
+        // Checks if some block in |called_function| has OpKill or OpUnreachable
+        // as its termination instruction.
+        bool hasKillOrUnreachableInstruction = false;
+        for (auto& called_function_block : *called_function) {
+          if (called_function_block.terminator()->opcode() == SpvOpKill ||
+              called_function_block.terminator()->opcode() ==
+                  SpvOpUnreachable) {
+            hasKillOrUnreachableInstruction = true;
+            break;
+          }
+        }
+
+        // |called_function| must not have a block with OpKill or OpUnreachable
+        // as its termination instruction.
+        if (hasKillOrUnreachableInstruction) {
+          continue;
+        }
+
+        function_call_instructions.push_back(&instruction);
+      }
+    }
+  }
+
+  for (auto& instruction : function_call_instructions) {
+    auto called_function = fuzzerutil::FindFunction(
+        GetIRContext(), instruction->GetSingleWordInOperand(0));
+
+    // Mapping the called function instructions.
+    std::map<uint32_t, uint32_t> result_id_map;
+    for (auto& called_function_block : *called_function) {
+      // The called function entry block label will not be inlined.
+      if (&called_function_block != &*called_function->entry()) {
+        result_id_map[called_function_block.GetLabelInst()->result_id()] =
+            GetFuzzerContext()->GetFreshId();
+      }
+
+      for (auto& instruction_to_inline : called_function_block) {
+        // The instructions are mapped to fresh ids.
+        if (instruction_to_inline.HasResultId()) {
+          result_id_map[instruction_to_inline.result_id()] =
+              GetFuzzerContext()->GetFreshId();
+        }
+      }
+    }
+
+    // Applies the inline function transformation.
+    ApplyTransformation(
+        TransformationInlineFunction(instruction->result_id(), result_id_map));
+  }
```

Comment:
It looks to me like the transformation might not be applicable by construction, in particular if the function call is not the penultimate instruction in its block and if the block terminator is not OpBranch.

To avoid this problem you should detect when this is the case, and apply a TransformationSplitBlock before applying the inlining transformation.